Riot/Modules/Room/DataSources/RoomDataSource.m

```diff
@@ -753,6 +762,40 @@ - (void)registerKeyVerificationTransactionNotification
                                                                        }];
 }
 
+- (void)registerBeaconInfoSummaryListner
+{
+    MXWeakify(self);
+    self.beaconInfoSummaryListener = [self.mxSession.aggregations.beaconAggregations listenToBeaconInfoSummaryUpdateInRoomWithId:self.roomId handler:^(id<MXBeaconInfoSummaryProtocol> beaconInfoSummary) {
+        MXStrongifyAndReturnIfNil(self);
+        @synchronized (self->bubbles)
+        {
+            [self refreshFirstCellWithBeaconInfoSummary:beaconInfoSummary];
+        }
+    }];
+}
+
+- (void)refreshFirstCellWithBeaconInfoSummary:(id<MXBeaconInfoSummaryProtocol>)beaconInfoSummary
+{
+    NSUInteger cellIndex = [bubbles indexOfObjectPassingTest:^BOOL(id<MXKRoomBubbleCellDataStoring>  _Nonnull cellData, NSUInteger idx, BOOL * _Nonnull stop) {
+        if ([cellData isKindOfClass:[RoomBubbleCellData class]])
+        {
+            RoomBubbleCellData *roomBubbleCellData = (RoomBubbleCellData*)cellData;
+            if ([roomBubbleCellData.beaconInfoSummary.id isEqualToString:beaconInfoSummary.id])
+            {
+                roomBubbleCellData.beaconInfoSummary = beaconInfoSummary;
```

Comment:
It's preferable to update `roomBubbleCellData` outside of `indexOfObjectPassingTest` scope but the result will be same.